README.md

```diff
@@ -73,24 +58,62 @@ Setup a environment variable called `MLIR_SYS_160_PREFIX` pointing to the mlir d
 MLIR_SYS_160_PREFIX=~/mlir
 ```
 
+### MacOS
+```bash
+brew install llvm@16
+export MLIR_SYS_160_PREFIX=/opt/homebrew/opt/llvm@16
+```
+
+## CLI Interface
+```
+Usage: cli --input <INPUT> <COMMAND>
+
+Commands:
+  compile  Compile to MLIR with LLVM dialect, ready to be converted by `mlir-translate --mlir-to-llvmir`
+  run      Compile and run a program. The entry point must be a function without arguments
+  help     Print this message or the help of the given subcommand(s)
+
+Options:
+  -i, --input <INPUT>  The input sierra file
+  -h, --help           Print help (see more with '--help')
+  -V, --version        Print version
+```
+
 ## Benchmarking
 
+### Requirements
+- [hyperfine](https://github.com/sharkdp/hyperfine): `cargo install hyperfine`
+- [cairo >=1.0](https://github.com/starkware-libs/cairo)
+- LLVM 16
+
+You need to setup some environment variables:
 ```bash
-cargo bench
+$S2M_BENCH_CAIRO_RUNNER=/path/to/cairo-run
+$MLIR_SYS_160_PREFIX=/path/to/llvm16
+```
+
+```bash
+make bench
 ```
 
 ## MLIR Resources
 - https://mlir.llvm.org/docs/Tutorials/
 
-## Translate output MLIR to LLVM IR
+## From MLIR to native binary
+```bash
+# to mlir with llvm dialect
+cargo r -- compile program.sierra -m --available-gas 9000000000 -o program.mlir
 
-```
-mlir-translate --mlir-to-llvmir output.mlir -o output.ll
+# translate mlir to llvm-ir
+mlir-translate --mlir-to-llvmir program.mlir -o program.ll
 
-# Compile with clang
-clang -O3 output.ll -o program
-./program
+# compile natively
+clang program.ll -Wno-override-module \
+    -L LLVM_DIR/lib -L"./target/release/" \
+    -lsierra2mlir_utils -lmlir_c_runner_utils \
+    -Wl,-rpath LLVM_DIR/lib \
+    -Wl,-rpath ./target/release/ \
+    -o program
```

Comment:
I think `LLVM_DIR` should be a variable `$LLVM_DIR` or just `$MLIR_SYS_160_PREFIX`. As it is now, it'll be treated as a string literal.